Assets/LeapMotionModules/InteractionEngine/Scripts/InteractionBehaviours/InteractionBehaviourBase.cs

```diff
@@ -450,6 +450,11 @@ public abstract class InteractionBehaviourBase : IInteractionBehaviour {
     #endregion
 
     #region UNITY MESSAGES
+
+    protected virtual void Awake() {
+      this.Reset();
```

Comment:
This will become problematic if the user extends the Reset functionality of their own implementation to do things that don't make sense to do in Awake.  There probably should be separate logic (maybe a different protected method even) that takes care of ensuring that the properties are valid.